api/osrd_infra/schemas/infra.py

```diff
@@ -124,71 +175,135 @@ def check_range(cls, v):
+    )
+    path: List[DirectionalTrackRange] = Field(description="List of the path corresponding to the routes")
 
 
 class SwitchPortConnection(BaseModel):
-    src: str
-    dst: str
-    bidirectional: bool
+    """
+    This class allows to know the connection between each port in switch type.
+    """
+
+    src: str = Field(description="Port name that is source of the connection")
+    dst: str = Field(description="Port name that is destination of the connection")
+    bidirectional: bool = Field(description="Know that the switch has connection in bidirectional or not")
 
 
 class SwitchType(BaseObjectTrait):
-    ports: List[str]
-    groups: Mapping[str, List[SwitchPortConnection]]
+    """
+    This class is used to define switch types.
+    We always have three switch types characterized by their identifier: point, cross and double cross.
+    """
+
+    ports: List[str] = Field(description="List of ports. A port correspond at the ends of the switches")
+    groups: Mapping[str, List[SwitchPortConnection]] = Field(description="Connection between and according ports")
 
 
 class Switch(BaseObjectTrait):
-    switch_type: ObjectReference
-    group_change_delay: float
-    ports: Mapping[str, TrackEndpoint]
-    label: str
+    """This class is used to define switches.
+    Switches are devices used for track changes."""
+
+    switch_type: ObjectReference = Field(description="Identifier and type of the switch type")
+    group_change_delay: float = Field(
+        description="Time it takes to change which group of the switch is activated", ge=0
+    )
+    ports: Mapping[str, TrackEndpoint] = Field(description="Location of differents ports according to track sections")
+    label: str = Field(description="Identifier of the switch")
 
 
 class TrackSectionLink(BaseObjectTrait):
-    src: TrackEndpoint
-    dst: TrackEndpoint
-    navigability: ApplicableDirections
+    """This class is used to define the track section links object.
+    Track section links is used to connect and link two track sections.
+    A track section link is characterized by its identifier.
+    """
+
+    src: TrackEndpoint = Field(
+        description="Starting track section, relative position in track section is defined by its end point"
+    )
+    dst: TrackEndpoint = Field(
+        description="Finish track section, relative position in track section is defined by its end point"
+    )
+    navigability: ApplicableDirections = Field(description="Direction of application of track section links")
 
 
 class SpeedSection(BaseObjectTrait):
-    speed_limit: Optional[float] = Field(description="Speed limit (m/s) applied by default to all trains")
+    """This class is used to define speed sections.
+    Speed sections is recognized by its identifier and are in meters per second."""
+
+    speed_limit: Optional[float] = Field(description="Speed limit (m/s) applied by default to all trains", gt=0)
```

Comment:
Not "to all trains" but rather "to trains that aren't in any specified category". It may be ignored if the speed for a category is higher.

At least that's how it's implemented in the back